### DISCUSSION TITLE
Changing Swell subdomain

### DISCUSSION THREAD

**Discussion by virajsoni06** [swellstores/community#250] [Help] [answered]
Hi, how can we change Swell's subdomain name? It shows up on hosted checkout to our customers. We had a demo store for testing and the url doesn't match our store's name and would like to change it

**Comment by sags95** [comment]
@virajsoni06 Do you have an existing swell-hosted storefront or are you using an externally hosted storefront?

**Reply by logeshswell** [reply]
@virajsoni06 You could add the subdomain directly from Swell dashboard following this link:
https://topmarks-test-store.swell.store/admin/storefronts/wWh1ZkPF?tab=domains
It would apply your domain changes for checkout as well.

**Reply by virajsoni06** [reply]
@sags95 @logeshswell we are using an externally hosted storefront

**Reply by virajsoni06** [reply]
We've added a new Storefront from Swell dashboard but when using Swell's checkout it shows the existing subdomain, `topmarks-test-store`, which we want to change

**Reply by virajsoni06** [reply]
@sags95 @logeshswell following up on this. How can we change it? Or can you change it from your end. This is quite critical for us before we launch the ecomm store

**Reply by logeshswell** [reply, marked as answer]
@virajsoni06 Can you follow the below instructions to set up a custom domain for your swell-hosted checkout?
1. Set [this storefront](https://topmarks-test-store.swell.store/admin/storefronts/wWh1ZkPF) as Primary and you could see **Domains** tab being shown next to **Navigation** tab.
2. Under Domains tab, click `Create domain` and follow the on-screen instructions.
3. Once the setup is successful, try creating a cart with some products and you should see the `checkout_url` with your subdomain instead of swell domain.
4. Finally, set your [custom storefront](https://topmarks-test-store.swell.store/admin/storefronts/IeqN2AAQ) as Primary.

**Reply by virajsoni06** [reply]
Thanks @logeshswell 

Added the domain and its verified but its not letting me enable auto-ssl. (Screenshot below)
Do I need to wait a while before I can enable SSL?

![Screenshot 2023-06-07 at 3 49 36 PM](https://github.com/swellstores/community/assets/13236286/55bbc3ef-e6dc-496b-a230-1d92d717c371)

**Reply by logeshswell** [reply]
@virajsoni06 I enabled auto-ssl for your custom domain now. Please check and confirm once.

**Reply by virajsoni06** [reply]
Hi @logeshswell, the checkout links don't seem to load at all now. 
fyi, the DNS is on CloudFlare and I've disabled proxy for the custom domain.

**Reply by virajsoni06** [reply]
![Screenshot 2023-06-08 at 5 13 43 PM](https://github.com/swellstores/community/assets/13236286/b859694c-a0cc-42ec-9188-ac013f13480f)

It ends up in 504 timeout

**Reply by logeshswell** [reply]
@virajsoni06 Sorry, there was a minor issue in the custom domain configuration and I fixed it.
I could access the below endpoint now, can you check and confirm once again?
https://store.exambeacon.co.uk/checkout/42894761d7c7b16dacfa071fee6add24

<img width="1512" alt="Screenshot 2023-06-08 at 7 15 18 PM" src="https://github.com/swellstores/community/assets/100766131/1f5a4a10-5211-4200-83aa-aa29368db1e4">

**Reply by virajsoni06** [reply]
Works now! Thanks @logeshswell
